### PR TITLE
Add the MPASJEDI 2024052700 case - CONUS12km with 30 ensembles

### PR DESCRIPTION
**Pull request by guoqing-noaa**
See issue #92 for more details

### CODE REVIEW

**Comment by SamuelDegelia-NOAA** [comment]
@guoqing-noaa Thanks for putting this case data together! It will be nice to have a much more realistic case to run MPAS experiments with. I have a couple of early questions here to open some discussion: 

1) Do we plan to keep the original `2022052619` case for running a CTest? Or should we maybe just take a small subset of the data added in this PR and rebuild the CTest? That way we aren't keeping multiple sets of test data around. 
2) What about making use of the existing `rrfs-test/` directory instead of `expr/` to include some of these scripts for setting up the experiment? 
3) Do we need to include these fix files (e.g., `geovars.yaml`, `namelist.atmosphere`, and `streams.atmosphere`) directly in the repo, or should we keep them in staged data that is copied in via `ush/init.sh`? For the existing CTest data we have been keeping those in the staged data. They are very small files but I think keeping them with the rest of the staged data makes sense because a user likely will not need to edit or look at them.

**Comment by ShunLiu-NOAA** [comment]
> @guoqing-noaa Thanks for putting this case data together! It will be nice to have a much more realistic case to run MPAS experiments with. I have a couple of early questions here to open some discussion:
> 
> 1. Do we plan to keep the original `2022052619` case for running a CTest? Or should we maybe just take a small subset of the data added in this PR and rebuild the CTest? That way we aren't keeping multiple sets of test data around.
Let's keep  `2022052619` case for a while until we fully practice the new test case. 
> 2. What about making use of the existing `rrfs-test/` directory instead of `expr/` to include some of these scripts for setting up the experiment?
We should keep the current infrastructure and use "rrfs-test/'.
> 3. Do we need to include these fix files (e.g., `geovars.yaml`, `namelist.atmosphere`, and `streams.atmosphere`) directly in the repo, or should we keep them in staged data that is copied in via `ush/init.sh`? For the existing CTest data we have been keeping those in the staged data. They are very small files but I think keeping them with the rest of the staged data makes sense because a user likely will not need to edit or look at them.
We should discuss this.

**Comment by guoqing-noaa** [comment]
Thanks @SamuelDegelia-NOAA  and @ShunLiu-NOAA for inputs.

> 1. Do we plan to keep the original `2022052619` case for running a CTest? Or should we maybe just take a small subset of the data added in this PR and rebuild the CTest? That way we aren't keeping multiple sets of test data around.

I agree with Shun that we can keep the 2022052619 case for a while. 
It will take a while to get the "2024052700" case more complete.

> 2. What about making use of the existing `rrfs-test/` directory instead of `expr/` to include some of these scripts for setting up the experiment?

My understanding of `rrfs_test/` is that it is used mostly for RRFS CTest. I might be wrong.
I can move expr to rrfs_test if no objections. 
However, I would like to skip the `setup_experiment.sh` step for the 2024052700 case. 
I am open to changes if anyone would like to make `setup_experiment.sh` work for this case. The bottom line is we would like to link data instead of copying data.

> 3. Do we need to include these fix files (e.g., `geovars.yaml`, `namelist.atmosphere`, and `streams.atmosphere`) directly in the repo, or should we keep them in staged data that is copied in via `ush/init.sh`? For the existing CTest data we have been keeping those in the staged data. They are very small files but I think keeping them with the rest of the staged data makes sense because a user likely will not need to edit or look at them.

Users will modify those files (I did) for different needs. It is good to include them in the repo directory. They usually will be included under parm/.

File: expr/mpasjedi_2024052700/colormap.py
Comment (delippi) [inline expr/mpasjedi_2024052700/colormap.py:1]: This is a duplicate of `rrfs-test/ush/colormap.py`. Maybe we could find a better place for it and then we can can just put it in PYTHONPATH.

File: expr/mpasjedi_2024052700/mpasjedi_increment_singleob.py
Comment (delippi) [inline expr/mpasjedi_2024052700/mpasjedi_increment_singleob.py:1]: This is also a duplicate script to `rrfs-test/ush/mpasjedi_increment_singleob.py`. If there were changes it might be best to just update the one in rrfs-test?

File: expr/mpasjedi_2024052700/testinput/msonet_airTemperature.yaml
Comment (delippi) [inline expr/mpasjedi_2024052700/testinput/msonet_airTemperature.yaml:1]: This is also a duplicate and probably not needed for the mpas case? This looks like an fv3jedi yaml.

File: expr/mpasjedi_2024052700/testinput/msonet_singleob_airTemperature_mpasjedi.yaml
Comment (delippi) [inline expr/mpasjedi_2024052700/testinput/msonet_singleob_airTemperature_mpasjedi.yaml:1]: This and the other yaml probably do not need the "singleob" tag in the filename. I know I did that a while ago but realized it wasn't needed because this singleob yaml should also work just fine for full ob test given your obs file has the correct name.

**Review by delippi — COMMENTED** [review]
Hi @guoqing-noaa, I think @SamuelDegelia-NOAA's suggestion of using rrfs-test is fine. While it is used for ctests, it is also used for setting up "rrfs tests" in general... At least that is my understanding. So that is why I had the setup_experiment script located there and it pulls the staged data, plotting scripts, etc. It would be nice to have this case ready to use for setup_experiment.sh. That setup script doesn't necessarily HAVE to copy data either, although that is/was just my preference since you mostly just copy data once and I like having hard copies incase anything went wrong. But it is understandable when datasets get too large.

Overall, it would be great to incorporate your init.sh into setup_experiment.sh and put things in rrfs-test. Idk if you've used setup_experiments.sh yet, but my intention was to make this as user friendly as possible when getting started. And it would be great to have one singular place where a new user can go and configure their experiments independent on which case or dycore they are working with.

File: expr/mpasjedi_2024052700/colormap.py
Comment (guoqing-noaa) [reply expr/mpasjedi_2024052700/colormap.py:1]: Yes, it is a duplicate. I can copy it from its original location.

File: expr/mpasjedi_2024052700/mpasjedi_increment_singleob.py
Comment (guoqing-noaa) [reply expr/mpasjedi_2024052700/mpasjedi_increment_singleob.py:1]: It is changed as the new mpasjedi case has different directory/name setting, such as different bkg name etc.
I think it is fine to have a variant copy here.

File: expr/mpasjedi_2024052700/testinput/msonet_airTemperature.yaml
Comment (guoqing-noaa) [reply expr/mpasjedi_2024052700/testinput/msonet_airTemperature.yaml:1]: Thanks, I can remove this one.

File: expr/mpasjedi_2024052700/testinput/msonet_singleob_airTemperature_mpasjedi.yaml
Comment (guoqing-noaa) [reply expr/mpasjedi_2024052700/testinput/msonet_singleob_airTemperature_mpasjedi.yaml:1]: I did not test this. I can change the name or remove it for now. We can add it back when we complete a test for the new case.

**Comment by guoqing-noaa** [comment]
@delippi Thanks for the review! 

I had an alternate view for `setup_experiment.sh` at the current stage. This script is very helpful for set up different experiments, FV3 or MPAS, hofx validation, etc. 

However, going forward, we will focus the MPAS JEDI case. There is no direct way to do hofx validation for MPASJEDI.
My expr setting enables a quick plug-in for anyone to try the MPASJEDI case (only need to specify the slurm account/queue/partition and it is done). Most developers don't need to deal with the FV3 core and the hofx validation any more.

On the other hand, `setup_experiment.sh` or a similar concept may be helpful when we work on the whole RRFSv2 workflow.

File: expr/mpasjedi_2024052700/mpasjedi_increment_singleob.py
Comment (delippi) [reply expr/mpasjedi_2024052700/mpasjedi_increment_singleob.py:1]: I think we should try to keep duplicate plot scripts to a minimum. We should be able to configure them to be used generically. You could also configure them to take arguments. I have examples. I'm not sure what the best solution is. At the same time, this case will replace the current MPAS case so we could just edit the script that already exists to be used for this case. It might be worth reconfiguring the MPAS ctests and everything based on this case and removing the current MPAS case completely since it is only good to test functionality.

File: expr/mpasjedi_2024052700/testinput/msonet_singleob_airTemperature_mpasjedi.yaml
Comment (delippi) [reply expr/mpasjedi_2024052700/testinput/msonet_singleob_airTemperature_mpasjedi.yaml:1]: Does this yaml use the built in single ob tool or does it just take an IODA file containing a single ob? If the latter then there is no need for the "singleob" tag in the filename. Let's remove this one for now and then consider renaming the other yaml with "singleob" in its name.

**Comment by delippi** [comment]
@guoqing-noaa, will we have a corresponding FV3JEDI and/or GSI case to go along with this new MPAS case? I'm under the impression or was at least hoping that we would. MPAS still needs validated somehow. While there is no way to do a direct comparison that we can make I think it would be useful to compare GSI with MPASJEDI to show that they agree to some degree expecting some differences. It seems dangerous to not have such validation.

**Comment by delippi** [comment]
Also, setup_experiment can be configured to turn off GSI and EVA. So it can just copy and setup your MPAS directory only.

File: expr/mpasjedi_2024052700/mpasjedi_increment_singleob.py
Comment (guoqing-noaa) [reply expr/mpasjedi_2024052700/mpasjedi_increment_singleob.py:1]: The current ctest mpas case is till valuable for providing a baseline test.
I think it is better to make a big update to current ctests after the summer when we will update to MPASv8.2 and MPASJEDIv3.0 (some more information in this issue #87 )

If you agree, we can make this version of py file the default one and users will need to modify settings if using it on the ctest mpas case.

File: expr/mpasjedi_2024052700/testinput/msonet_singleob_airTemperature_mpasjedi.yaml
Comment (guoqing-noaa) [reply expr/mpasjedi_2024052700/testinput/msonet_singleob_airTemperature_mpasjedi.yaml:1]: Thanks!

File: expr/mpasjedi_2024052700/mpasjedi_increment_singleob.py
Comment (delippi) [reply expr/mpasjedi_2024052700/mpasjedi_increment_singleob.py:1]: I agree that the current MPAS case can provide some good baseline tests, but at the same time it might be good to do some house keeping. It might take a little bit of effort to reconfigure all the ctests with the new case, but it might be better long term to only have the one case. I'm also okay with doing this after the summer with those updates though.

I'm not currently using the mpas version of the python script. I'm not sure if anyone else is either since that case really isn't too helpful other than for ctests. I'm okay with you updating the one that we currently have.

**Comment by guoqing-noaa** [comment]
> @guoqing-noaa, will we have a corresponding FV3JEDI and/or GSI case to go along with this new MPAS case? I'm under the impression or was at least hoping that we would. MPAS still needs validated somehow. While there is no way to do a direct comparison that we can make I think it would be useful to compare GSI with MPASJEDI to show that they agree to some degree expecting some differences. It seems dangerous to not have such validation.

Unfortunately, we cannot do direct comparisons between MPASJEDI and GSI like we can do between FV3JEDI and GSI. We have to live with this reality.

But we still manage to have some baseline GSI tests to compare with MPASJEDI.
@Junjun-NOAA and @spanNOAA has been exploring to establish some comparison capability. There are two routes to go. We can discuss this further in our joint meeting. Or we can create an issue now to update the progress. Thanks for the suggestion!

**Comment by guoqing-noaa** [comment]
@SamuelDegelia-NOAA @delippi @ShunLiu-NOAA 

Thanks all for the review!
I've made updates following suggestions. Welcome your second round of review. Thanks!

**Comment by guoqing-noaa** [comment]
I should note that `rrfs-test/ush/mpasjedi_spread.py` comes from @SamuelDegelia-NOAA 
I just added it to the repo.

**Comment by delippi** [comment]
Hi @guoqing-noaa, thanks for making some changes. I would still say that your link_expr script could be incorporated into the setup_experiment script. Then the run_jedi script should go rrfs-test/scripts/templates--well, just update the one that is already there. You could do the same thing with run_bump script. The namelist and streams files could either go in the staged location or not sure where. Then we wouldn't need this additional mpasjedi_2024052700 directory (that directory would be made as a work space after running the setup_experiment script). Basically I'm trying to make sure we don't have too many scripts that do similar things--if they can be combined I would like to see them combined.

The setup_experiment script doesn't necessarily only have to be for hofx validation. I have options to configure so that you don't have to copy GSI or EVA. So this script could be used to setup your MPAS case. Also, you could add some logic that if it is the MPAS case, then just link instead of copy/rsync. The old MPAS case could be commented out and replaced with this case in that script.

I'm happy to discuss further.

**Comment by guoqing-noaa** [comment]
@delippi Thanks for the reivew. It will be good if you can help with adopting `setup_experiment `. I am open to any changes.

**Comment by delippi** [comment]
@guoqing-noaa, sure. I will help.

**Comment by guoqing-noaa** [comment]
Thanks, @delippi!

I would provide a second opinion. Most people will use this MPASJEDI case to do JEDI exploration and only a very few developers may use the FV3JEDI case or the hofx validation to examine some mysteries if we are not able to get a clear answer when working on MPASJEDI. So my current setting is much easier and simpler for them.

**Comment by guoqing-noaa** [comment]
A few suggestions on the `setup_experiment ` while you are working on it:

1. A script can detect the RDASApp directory automatically, user don't need to manually set it. You may refer the run_mpasjedi.sh in this PR.

2. I suggest we use the following default setting:
```
DYCORE="MPAS" #FV3 | MPAS
GSI_TEST_DATA="NO"
YOUR_PATH_TO_GSI="/path/to/your/installation/of/GSI" # Not needed for MPASJEDI
EVA="NO"
```

3. Users will still need to modify the final job card to change partition, qos, or reservation in current `setup_experiment ` configuration

**Comment by guoqing-noaa** [comment]
@delippi @SamuelDegelia-NOAA @TingLei-NOAA @ShunLiu-NOAA 
I think this PR is good to go. Thank you for the help, @delippi!

**Comment by TingLei-NOAA** [comment]
@guoqing-noaa  and/or others, A general quesiton. When I check out rdasapp with your PR,  what scripts/commands shoud I use to set up and run the mpasjedi case added in this PR?

**Comment by guoqing-noaa** [comment]
@delippi Are you able to help @TingLei-NOAA to run the MPAS2024052700 case? My internet connection is very slow. It will be great if you can help. Thanks a lot!

**Comment by delippi** [comment]
> @guoqing-noaa and/or others, A general quesiton. When I check out rdasapp with your PR, what scripts/commands shoud I use to set up and run the mpasjedi case added in this PR?

@TingLei-NOAA, please use `rrfs-test/scripts/setup_experiment.sh`. This script is not only for the hofx validation but also for setting up experiments in general. Just make sure to have dycore="MPAS". The GSI and EVA settings can be turned off. These should all be the default settings now.

**Comment by TingLei-NOAA** [comment]
@delippi  Thanks.  Using the setup_experiment.sh, it successfully created a exp dir.  The changed script is /scratch2/NCEPDEV/fv3-cam/Ting.Lei/dr-guoqing/RDASApp//rrfs-test/scripts/setup_experiment.sh. 
The expdir is dr-test . But running "sbatch run_mpasjedi.sh sonde_singeob_airTemperature_mpasjedi.yaml"  got a failed run with error messages like 
```
ERROR: In file ./streams.atmosphere, definition of stream "background" references file stream_list/stream_list.atmosphere.background that cannot be opened for reading.
```
I guess some files are not copied to this expdir. 
I will leave the fix to @guoqing-noaa  when he is available or other rdasapp developers.

**Comment by delippi** [comment]
Hey @TingLei-NOAA, I think I know what the issue is. I made a patch to @guoqing-noaa's PR but was unsure about one part so this part was left out. I think I understand how that was supposed to work now. I will try to make another patch to the PR.

**Comment by guoqing-noaa** [comment]
@ShunLiu-NOAA @delippi Do we have any remaining questions on this PR? It will be best to merge this PR as earlier as possible. A few GSL developers would like to use this case to test/do some developments. Thanks!

**Comment by SamuelDegelia-NOAA** [comment]
@guoqing-noaa It sounds like we were having some issues running the experiments per @TingLei-NOAA's recent comment. Are you able to merge @delippi's PR in your fork so that we could test it again?

**Comment by guoqing-noaa** [comment]
Thanks, @SamuelDegelia-NOAA!
Let me work on this.

**Comment by guoqing-noaa** [comment]
@delippi Thanks for your new PR to my branch! We will need to have a tag up to resolve differences.
Generally, `init.sh `is not only used by the MPAS JEDI case. It has more general use and will have more functionalities. We would like to keep it. I can discuss this more with you next week.

At the same time, I will share my personal branch to those who would like to try the new MPAS case as soon as possible.
Once this PR is resolved sometime next week, we can transition to your setup_expr method. Thanks!

**Comment by TingLei-NOAA** [comment]
@guoqing-noaa ,so for the current PR, do we have a way to set up you case or shall we wait for a new update?

**Comment by ShunLiu-NOAA** [comment]
@guoqing-noaa Do you have a chance to look at Ting's question? Ting will test the case for reviewing this PR.

**Comment by guoqing-noaa** [comment]
> @guoqing-noaa ,so for the current PR, do we have a way to set up you case or shall we wait for a new update?

Hi @TingLei-NOAA , I sent you an email last Friday on how to test this MPASJEDI case. I will reply to that email to remind you.
For this PR, I will think more about how to resolve the differences with @delippi

**Comment by TingLei-NOAA** [comment]
@guoqing-noaa   Thanks for this clarification!

**Comment by guoqing-noaa** [comment]
Per Donnie's test, the PR now works correctly.  
@ShunLiu-NOAA @delippi @SamuelDegelia-NOAA @TingLei-NOAA Please make a new review. Thanks!

File: rrfs-test/scripts/setup_experiment.sh
Comment (SamuelDegelia-NOAA) [inline rrfs-test/scripts/setup_experiment.sh:137]: Could we add this back in? The job script `run_mpasjedi.sh` checks for the `testinput` directory if a yaml file is not passed into the job script. Plus it helps to have all of those yamls in the run directory if we want to try different configurations.

**Review by TingLei-NOAA — DISMISSED** [review]
The changes look good to me when it had been tested by @delippi as @guoqing-noaa mentioned. 
Thanks!

File: rrfs-test/ush/mpasjedi_increment_singleob.py
Comment (SamuelDegelia-NOAA) [inline rrfs-test/ush/mpasjedi_increment_singleob.py:44]: Could we make the same edits in `mpasjedi_increment_fulldom.py`? I am fine with renaming the output file from `an.` to `anl.` but now the `mpasjedi_increment_fulldom.py` script will not work without changing these corresponding lines in it.

**Comment by SamuelDegelia-NOAA** [comment]
Overall things are working for me and look good after removing the Jet-specific job submission lines above. I ran the EnVar single observation test and the results look more realistic than our previous case. Looking forward to working on this case once the PR is merged! 

Apart from my other line-specific comments, I have a couple other suggestions. Apologies if these have already been discussed offline with @delippi. 

1. Could we add something to the wiki or build instructions about needing to run `ush/init.sh` to set up this case? Just running `setup_experiment.sh` alone will not work unless the former is run first. An alternative is to integrate `ush/init.sh` into `setup_experiment.sh`. I know you mentioned earlier about not wanting to do this, but with how fast it runs (nearly instant) I still think it could be a good idea to save a step. 

1. Do the `namelist.atmosphere` and `streams.atmosphere` files need to go in `rrfs-test/testinput`? We have mostly been housing JEDI yamls in there. Could we instead put those in the new `fix/` dir created by `ush/init.sh`?

**Comment by guoqing-noaa** [comment]
> Overall things are working for me and look good after removing the Jet-specific job submission lines above. I ran the EnVar single observation test and the results look more realistic than our previous case. Looking forward to working on this case once the PR is merged!
> 
> Apart from my other line-specific comments, I have a couple other suggestions. Apologies if these have already been discussed offline with @delippi.
> 
> 1. Could we add something to the wiki or build instructions about needing to run `ush/init.sh` to set up this case? Just running `setup_experiment.sh` alone will not work unless the former is run first. An alternative is to integrate `ush/init.sh` into `setup_experiment.sh`. I know you mentioned earlier about not wanting to do this, but with how fast it runs (nearly instant) I still think it could be a good idea to save a step.
> 2. Do the `namelist.atmosphere` and `streams.atmosphere` files need to go in `rrfs-test/testinput`? We have mostly been housing JEDI yamls in there. Could we instead put those in the new `fix/` dir created by `ush/init.sh`?

Good comments! Thanks @SamuelDegelia-NOAA 

1. Sorry that I actually should commit init.sh first and not mix it with this PR. 
init.sh is to initialize fix/ directory for a centralized data location and create a pre-commit hook to prevent accidental commit of binary data into the repo. Its final form will be more similar to this:
https://github.com/rrfs2/rrfs-workflow/blob/main/ush/init.sh
init.sh usually runs at the building time (this is what rrfsv1 did). But people may skip the building process. To prevent this, I would suggest adding `init.sh` into `setup_experiment.sh` as a solution as well. @delippi are you okay with that?
I understand you mentioned corner cases to me where one may not be able to run `init.sh`. Let's ignore corner cases for now?

2. I am fine to put them under parm/ (namelists usually go into parm/).

Thanks!

**Comment by delippi** [comment]
@guoqing-noaa & @SamuelDegelia-NOAA, I think `init.sh` should just be part of the build process. The path `YOUR_PATH_TO_RDASAPP` in `setup_experiment.sh` should only point to an RDASApp that has already been built. I hope that answers the question? I didn't have any such problem because I pointed to Guoqing's RDASApp build and it worked fine for me.

**Comment by SamuelDegelia-NOAA** [comment]
I think integrating `init.sh` into the build process is also a good solution.

**Comment by SamuelDegelia-NOAA** [comment]
Hi @guoqing-noaa, I am trying to run some LETKF tests with the new 2024052700 case here. But I am getting an error with missing pressure data in the ensemble files. It looks like the only pressure variable in the ensemble files is `surface_pressure`. The `pressure_p` and `pressure_base` variables are present the the deterministic background file for your new case, but the LETKF also needs these variables in the ensemble members. Is it possible to add this?

**Comment by guoqing-noaa** [comment]
@SamuelDegelia-NOAA Thanks for testing this.
Yes, this is an issue. We created a ticket here:
https://github.com/rrfs2/rrfs-workflow/issues/22

I am not sure who would like and will be able to work on this.

For RDASApp,  we can update the ensemble BEC by using 6h MPAS forecasts. But that will also have a problem that we don't add any perturbations to the 30 members yet. @chunhuazhou is working on applying realistic ensemble initialization. Her work may provide some realistic ensemble BEC.

**Comment by SamuelDegelia-NOAA** [comment]
@guoqing-noaa I see, so the deterministic background is a 6h MPAS forecast which adds the pressure variables? But the ensemble members are cold start files that do not have it yet? If I am understanding that correctly, could we just run forecasts with the ensemble to generate the pressure fields?

**Comment by guoqing-noaa** [comment]
@SamuelDegelia-NOAA 
Yes, if you run ensemble forecasts, we will get pressure fields. We have had that before. But it gives unexpected small single ob increments so we changed them.

**Comment by SamuelDegelia-NOAA** [comment]
@guoqing-noaa Ahh okay, so that is the issue we found in #89. Do you happen to still have those ensemble forecasts saved? I would be happy to work on adding random perturbations to them to increase spread, just so we could get the LETKF working for this new test case.

**Comment by guoqing-noaa** [comment]
@SamuelDegelia-NOAA You are right. It is the issue in #89.
If you can add random perturbation, that would be great! Thanks!

The 30 member 6h forecasts are here
`/scratch1/BMC/wrfruc/gge/RDASApp/expr/jedi-assim_test/rrfs-data_mpasjedi_2024052700/Data/ens.czhou.00z`

**Comment by guoqing-noaa** [comment]
@SamuelDegelia-NOAA FYI, @chunhuazhou is working on updating the ensembles in this case to be 1h MPAS forecasts initialized from GEFS 29h fcst (Per David Dowell and Chunhua's previous examination, GEFS 30 h fcsts have better ensemble spread).

**Comment by SamuelDegelia-NOAA** [comment]
@guoqing-noaa thank you for the update. I did add perturbations to the forecast ensemble to be usable with LETKF, but I've been running into this new issue with the hofx for each member being the exact same (https://github.com/NOAA-EMC/RDASApp/issues/111). I'm working on debugging that now. I'd assume the new forecasts @chunhuazhou is working on will have the same problem, but we'll see.

**Comment by guoqing-noaa** [comment]
@ShunLiu-NOAA @TingLei-NOAA @delippi @SamuelDegelia-NOAA @hu5970  Please review the updates.

The ensembles have been updated to use the fhr35 GEFS initialized 1-h MPAS forecasts prepared by @chunhuazhou .  
The fix/ directory has been added (this will facilitate a centralized location for RDAS_DATA, I will create a new PR for this).   
A quick start option has been added. Donnie and I previously agreed that we keep both the powerful setup_experiment.sh and this quick start option. But I did not get time to work on this until now.

Thanks and let me know if you have any concerns.

**Comment by SamuelDegelia-NOAA** [comment]
I tested LETKF using the new ensembles and things are looking good. Thanks @chunhuazhou for adding these. Increments for a single observation test are about double the magnitude compared to the old ensemble using the same obs error and innovation.

**Comment by guoqing-noaa** [comment]
@SamuelDegelia-NOAA Thanks for testing!
Yes, the ensemble spread looks better using the 35h GEFS forecasts. This is copied from issue #89 

level=23, 20240527_00z ensemble spread from 1h MPAS fcst initialized from 35h GEFS forecast

image
    
    
The associate temperature increment at level 23:
    
image

**Comment by guoqing-noaa** [comment]
@ShunLiu-NOAA @TingLei-NOAA @delippi @SamuelDegelia-NOAA @hu5970
Let me know if you have any concerns.

File: .gitignore
Comment (TingLei-NOAA) [inline .gitignore:155]: @guoqing-noaa  Would you explain what the purposes are for use of .agent dir ?

**Review by SamuelDegelia-NOAA — DISMISSED** [review]

File: rrfs-test/scripts/setup_experiment.sh
Comment (guoqing-noaa) [reply rrfs-test/scripts/setup_experiment.sh:137]: @SamuelDegelia-NOAA I think I did not change this file. @delippi Could you help address Sam's question here?

File: rrfs-test/ush/mpasjedi_increment_singleob.py
Comment (guoqing-noaa) [reply rrfs-test/ush/mpasjedi_increment_singleob.py:44]: Thanks, @SamuelDegelia-NOAA What change we should make? Do you mean we need to make this file also work for the previous MPASJEDI case? 

I have suggested I keep a standalone copy of this file for the new MPASJEDI case. But after discussion with Donnie, we went another route.  I think it is still a good idea  that we don't change the behavior of the previous MPASJEDI case. I will make a change and keep two copies of this file. Does this work for you?

Comment (SamuelDegelia-NOAA) [reply rrfs-test/ush/mpasjedi_increment_singleob.py:44]: Sorry @guoqing-noaa, I think my review comments just showed up despite typing them a couple of weeks ago because I still had my review as "pending". I am still learning how to use Github, hah. 

But anyways I just meant to also edit `mpasjedi_increment_fulldom.py` to be consistent with these changes to  `mpasjedi_increment_singleob.py` for the new MPAS-JEDI case. This new case changes the analysis filename prefix from `an.${DATE}` to `ana.${DATE}` and changes the path for these static files. This change to the singleob plotting script highlighted here represents those new file names, but `mpasjedi_increment_fulldom.py` still uses the original names. So basically I am just asking to make these same three line changes to `mpasjedi_increment_fulldom.py`.

Comment (guoqing-noaa) [reply rrfs-test/ush/mpasjedi_increment_singleob.py:44]: Ah, I got it. Thanks for the clarification and mentioning this. I forgot that part.

File: .gitignore
Comment (guoqing-noaa) [reply .gitignore:155]: > @guoqing-noaa Would you explain what the purposes are for use of .agent dir ?

@TingLei-NOAA Good question!  The .agent link is a byproduct of the fix file management which was used in the current rrfsv1 workflow.  

The general idea is that binary fix files and text fix files coexist under the fix/ directory. Text files are tracked by the repo similar as other script/config/etc files while binary files are tracked differently using links which point to the real data locations which is given by the RDAS_DATA variable. In different machines, RDAS_DATA takes different values. It will be tedious if we change every links when transitioning from one machine to another machine. The .agent link is used to simplify this task and we only need to update the .agent link on different machines. Here is a diagram to show this. [This link](https://github.com/rrfs2/rrfs-workflow/wiki/fix-file-management-policy) provides more details on the fix file management. Let me know if you want to have a short tagup to have a further discussion on this.

image

**Comment by guoqing-noaa** [comment]
@SamuelDegelia-NOAA I have updated the mpasjedi_increment_fulldom.py file as suggested and I can generated a figure as follows.  
Sorry that this commit automatically dismissed your approval. Could you review again? Thanks!
    
image

**Review by SamuelDegelia-NOAA — APPROVED** [review]
Thanks for the quick update @guoqing-noaa!

File: .gitignore
Comment (guoqing-noaa) [reply .gitignore:155]: @TingLei-NOAA we can talk more on this but I think it is not a roadblock to merge this PR. I will resolve this conversation to get ready for a merge. Thanks!

File: rrfs-test/scripts/setup_experiment.sh
Comment (guoqing-noaa) [reply rrfs-test/scripts/setup_experiment.sh:137]: I will leave this to @delippi

File: .gitignore
Comment (TingLei-NOAA) [reply .gitignore:155]: @guoqing-noaa  I think this PR don't have to have my approval to get merged . But from my point of view, it is a question to be clarified before merging. So, I suggest  to leave it unresolved as it is ( and, I think it is not required that every conversion should be resolved before a PR's merging).  Thanks.

**Comment by guoqing-noaa** [comment]
I plan to merge this PR early tomorrow if no objections. There may be a few minor things but we can always update it and make it better later. A few is awaiting to use changes in this PR (they have been using my forked branch) and it is better to use the official RDASApp.

File: .gitignore
Comment (guoqing-noaa) [reply .gitignore:155]: @TingLei-NOAA  Sorry that we have to resolve all conversations otherwise a merge is NOT allowed. 
But even the conversation is resolved at the PR level, we can continue to make comments, raise concerns and even we can create a new issue for this.

**Comment by guoqing-noaa** [comment]
I had a tag-up meeting with @TingLei-NOAA this morning. We discussed and addressed the remaining concerns. I think it is good to merge this PR now.

**Review by TingLei-NOAA — APPROVED** [review]

**Review by ShunLiu-NOAA — APPROVED** [review]

**Comment by ShunLiu-NOAA** [comment]
@guoqing-noaa is it possible that you stage the data and test case to role account on Hera and Jet before we merge this PR? We can also have another PR to move test cases and data to role account. Which one is better for you now?

**Comment by guoqing-noaa** [comment]
@ShunLiu-NOAA all data has been staged under the role account role.rrfs-fv3-cam on Hera.

**Pull request merged**
